compiler-core/src/parse.rs

```diff
@@ -3139,37 +3153,25 @@ where
     ) -> Result<u32, ParseError> {
         let (_, name, _) = self.expect_name()?;
 
-        match name.as_str() {
-            "erlang" => {
-                let _ = self.expect_one(&Token::Comma)?;
-                let (_, module, _) = self.expect_string()?;
-                let _ = self.expect_one(&Token::Comma)?;
-                let (_, function, _) = self.expect_string()?;
-                let _ = self.maybe_one(&Token::Comma);
-                let (_, end) = self.expect_one(&Token::RightParen)?;
-                if attributes.external_erlang.is_some() {
-                    return parse_error(ParseErrorType::DuplicateAttribute, SrcSpan { start, end });
-                }
-                attributes.external_erlang = Some((module, function));
-                Ok(end)
-            }
+        let target = match name.as_str() {
+            "erlang" => Target::Erlang,
+            "javascript" => Target::JavaScript,
+            _ => return parse_error(ParseErrorType::UnknownAttribute, SrcSpan::new(start, end)),
+        };
 
-            "javascript" => {
-                let _ = self.expect_one(&Token::Comma)?;
-                let (_, module, _) = self.expect_string()?;
-                let _ = self.expect_one(&Token::Comma)?;
-                let (_, function, _) = self.expect_string()?;
-                let _ = self.maybe_one(&Token::Comma);
-                let _ = self.expect_one(&Token::RightParen)?;
```

Comment:
Here's the subtle bug: we weren't pulling off the `end` for the RightParen token in this branch, but we were in the erlang branch. I decided to commonize as much of these two branches as possible to avoid such inconsistencies.